cli_tools/osconfig_agent/main.go

```diff
@@ -42,22 +101,42 @@ func strIn(s string, ss []string) bool {
 	return false
 }
 
-func main() {
-	flag.Parse()
-	patchInit()
-
-	ctx := context.Background()
-
+func run(ctx context.Context) {
 	client, err := osconfig.NewClient(ctx, option.WithEndpoint(*endpoint), option.WithCredentialsFile(*oauth))
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("NewClient Error:", err)
 	}
 
-	res, err := lookupConfigs(ctx, client, *resource)
+	res, err := getResource(*resource)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("getResource error:", err)
 	}
-	patchManager(res.PatchPolicies)
 
-	//runUpdates()
+	patchInit()
+	ticker := time.NewTicker(interval)
+	for {
+		res, err := lookupConfigs(ctx, client, res)
```

Comment:
You you mean to reuse `res` here? Does `lookupConfigs()` and `getResource()` return the same thing?

Reply:
No, it doesn't matter as this is in a different scope but I renamed to avoid confusion.